pyuvdata/uvdata/mir_parser.py

```diff
@@ -225,696 +42,1062 @@ class MirParser(object):
-        filepath : str
-            Path to the folder containing the mir data set.
-        in_start_dict: dict of tuples
-            Dictionary returned from scan_int_start, which records position and
-            record size for each integration
-        sp_data : ndarray of sp_data_type
-            Array from "sp_read", returned by "read_sp_read".
-
-        Returns
-        -------
-        vis_list : list of ndarrays
-            List of ndarrays (dtype=csingle/complex64), with indices equal to sphid
-            and values being the floating-point visibilities for the spectrum
-        """
-        # Gather the needed metadata
-        inhid_arr = sp_data["inhid"]
-        nch_arr = sp_data["nch"]
-        dataoff_arr = sp_data["dataoff"] // 2
-        sp_pos = np.arange(len(dataoff_arr))
-
-        unique_inhid = np.unique(inhid_arr)
-        unique_nch = np.unique(nch_arr)
-        vis_list = []
-        sp_pos_list = []
-        for inhid in unique_inhid:
-            packdata = MirParser.read_vis_data(filepath, {inhid: in_start_dict[inhid]})[
-                inhid
-            ]["packdata"]
-            data_mask = inhid_arr == inhid
-            dataoff_subarr = dataoff_arr[data_mask]
-            nch_subarr = nch_arr[data_mask]
-            sp_pos_subarr = sp_pos[data_mask]
-            scale_fac = np.power(2.0, packdata[dataoff_subarr], dtype=np.float32)
-            for spec_size in unique_nch:
-                start_list = dataoff_subarr[spec_size == nch_subarr] + 1
-                end_list = start_list + (spec_size * 2)
-                temp_data = np.multiply(
-                    scale_fac[spec_size == nch_subarr, None],
-                    np.array(
-                        [packdata[idx:jdx] for idx, jdx in zip(start_list, end_list)]
-                    ),
-                    dtype=np.float32,
-                ).view(dtype=np.complex64)
-                vis_list.extend([temp_data[idx, :] for idx in range(len(start_list))])
-                # Record where the data _should_ go in the list
-                sp_pos_list.extend(sp_pos_subarr[spec_size == nch_subarr])
-        # Do a bit of order swapping so that things match with sp_read
-        sp_pos_dict = {sp_pos_list[idx]: idx for idx in range(len(sp_pos_list))}
-        vis_list = [vis_list[sp_pos_dict[idx]] for idx in range(len(sp_pos_list))]
-        return vis_list
+        return int_dict
 
-    @staticmethod
-    def parse_raw_data(filepath, in_start_dict, sp_data):
+    def _fix_int_dict(self, data_type):
         """
-        Read "sch_read" mir file into a python dictionary (@staticmethod).
+        Fix an integration postion dictionary.
 
-        Note that this returns a list rather than a dict to help expedite some
-        processing tasks where order of the arrays matters.
+        Note that this function is not meant to be called by users, but is instead
+        part of the low-level API for handling error correction when reading in data.
```

Comment:
again, maybe don't use the term "API" for methods that start with underscores. Maybe "helper"?

Reply:
Fixed in a2f46b4b, also made this a non-public function.